### PR TITLE
Add macOS for Docker build in CI

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
*Issue #, if available:*

*Description of changes:*

### CODE REVIEW

**Review by john-h-kastner-aws — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,9 +47,16 @@ jobs:
...
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest-large
+    runs-on: ${{ matrix.os }}
     name: Test Docker build
-    runs-on: ubuntu-latest
     steps:
+      - name: Set up Docker
+        uses: crazy-max/ghaction-setup-docker@v3
```
Comment (john-h-kastner-aws) [inline .github/workflows/ci.yml:59]: Is there any potential issue with trusting  this `crazy-max`?

Comment (shaobo-he-aws) [reply .github/workflows/ci.yml:59]: Good question. It should be fine since we're not using any private keys?

Comment (shaobo-he-aws) [reply .github/workflows/ci.yml:59]: Let me try to get rid of it.

**Review by khieta — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (shaobo-he-aws) [reply .github/workflows/ci.yml]: I managed to do it but build on mac is taking a long time. Optimizing.

**Review by andrewmwells-amazon — APPROVED** [review]

**Comment by shaobo-he-aws** [comment]
Closed this PR because docker on macOS is unstable (at least on GitHub action images).

**Pull request closed without merging**